# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1265090011 18000
#      Tue Feb 02 00:53:31 2010 -0500
# Node ID fa7b5751730cb6f36562d4c0bd558fce5430f128
# Parent  e45ccac6017f0d28662e5f4247215258b2f6d261
use gnulib time, sys_time, and sys_times modules

diff --git a/ChangeLog b/ChangeLog
--- a/ChangeLog
+++ b/ChangeLog
@@ -1,8 +1,16 @@
+2010-02-02  John W. Eaton  <jwe@octave.org>
+
+	* bootstrap.conf (gnulib_modules): Include sys_time, sys_times,
+	and time in the list.
+	* configure.ac: Don't check for sys/time.h, sys/times.h,
+	sys/stat.h, sys/utimes, or utimes.h.
+	Don't call AC_HEADER_TIME.
+
 2010-02-02  John W. Eaton  <jwe@octave.org>
 
 	* acinclue.m4 (OCTAVE_STRPTIME_BROKEN): Delete.
 	* configure.ac: Don't call it.
 
 2010-02-01  John W. Eaton  <jwe@octave.org>
 
 	* acinclude.m4 (OCTAVE_PROG_MAKEINFO): Delete macro.
diff --git a/bootstrap.conf b/bootstrap.conf
--- a/bootstrap.conf
+++ b/bootstrap.conf
@@ -32,16 +32,19 @@ gnulib_modules="
   readlink
   rename
   rmdir
   stat
   strftime
   strptime
   symlink
   sys_stat
+  sys_time
+  sys_times
+  time
   times
   unistd
   unlink
   round
 "
 
 # Additional xgettext options to use.  Use "\\\newline" to break lines.
 XGETTEXT_OPTIONS=$XGETTEXT_OPTIONS'\\\
diff --git a/configure.ac b/configure.ac
--- a/configure.ac
+++ b/configure.ac
@@ -1446,27 +1446,25 @@ OCTAVE_FAST_INT_OPS
 ### Check for long double type (for 64-bit integers)
 
 AC_CHECK_SIZEOF(long double)
 
 ### Checks for header files.
 
 AC_HEADER_STDC
 AC_HEADER_DIRENT
-AC_HEADER_TIME
 AC_HEADER_SYS_WAIT
 
 ### C headers
 
 AC_CHECK_HEADERS(curses.h direct.h dlfcn.h fcntl.h \
   floatingpoint.h grp.h ieeefp.h inttypes.h locale.h memory.h nan.h \
   ncurses.h poll.h pthread.h pwd.h stdint.h sunmath.h sys/ioctl.h \
-  sys/param.h sys/poll.h sys/resource.h sys/select.h sys/stat.h \
-  sys/time.h sys/times.h sys/utsname.h sys/utime.h termcap.h \
-  utime.h)
+  sys/param.h sys/poll.h sys/resource.h sys/select.h \
+  sys/utsname.h termcap.h)
 
 ### C++ headers
 
 AC_LANG_PUSH(C++)
 AC_CHECK_HEADERS(sstream)
 AC_LANG_POP(C++)
 
 have_termios_h=no
diff --git a/liboctave/ChangeLog b/liboctave/ChangeLog
--- a/liboctave/ChangeLog
+++ b/liboctave/ChangeLog
@@ -1,8 +1,17 @@
+2010-02-02  John W. Eaton  <jwe@octave.org>
+
+	* oct-time.h: Include <ctime>, not "systime.h".
+	* oct-time.cc: Include <strftime.h>, not "strftime.h".
+	Don't include <windows.h>.
+
+	* systime.h: Delete.
+	* Makefile.am (INCS): Remove it from the list.
+
 2010-02-02  John W. Eaton  <jwe@octave.org>
 
 	* lo-utils.h, lo-cutils.c (oct_strptime): Delete.
 	* oct-time.cc (octave_strptime::init): Call strptime, not oct_strptime.
 
 2010-01-28  Jaroslav Hajek  <highegg@gmail.com>
 
 	* dMatrix.cc (Matrix::determinant): Treat matrices marked as singular
diff --git a/liboctave/Makefile.am b/liboctave/Makefile.am
--- a/liboctave/Makefile.am
+++ b/liboctave/Makefile.am
@@ -238,17 +238,16 @@ INCS = \
   randpoisson.h \
   regex-match.h \
   sparse-sort.h \
   sparse-util.h \
   statdefs.h \
   str-vec.h \
   sun-utils.h \
   sysdir.h \
-  systime.h \
   syswait.h \
   $(MATRIX_INC)
 
 PRIVATE_INCS = \
   oct-glob.h
 
 OTHER_INC = \
   intNDArray.cc \
diff --git a/liboctave/oct-time.cc b/liboctave/oct-time.cc
--- a/liboctave/oct-time.cc
+++ b/liboctave/oct-time.cc
@@ -22,28 +22,21 @@ along with Octave; see the file COPYING.
 
 #ifdef HAVE_CONFIG_H
 #include <config.h>
 #endif
 
 #include <climits>
 #include <ctime>
 
+#include <sys/time.h>
 #include <sys/types.h>
 #include <unistd.h>
 
-#if defined (OCTAVE_USE_WINDOWS_API)
-#define WIN32_LEAN_AND_MEAN
-#include <windows.h>
-#undef min
-#undef max
-#endif
-
-#include <sys/time.h>
-#include "strftime.h"
+#include <strftime.h>
 
 #include "lo-error.h"
 #include "lo-math.h"
 #include "lo-utils.h"
 #include "oct-time.h"
 
 octave_time::octave_time (const octave_base_tm& tm)
 {
diff --git a/liboctave/oct-time.h b/liboctave/oct-time.h
--- a/liboctave/oct-time.h
+++ b/liboctave/oct-time.h
@@ -18,20 +18,20 @@ You should have received a copy of the G
 along with Octave; see the file COPYING.  If not, see
 <http://www.gnu.org/licenses/>.
 
 */
 
 #if !defined (octave_time_h)
 #define octave_time_h 1
 
+#include <ctime>
 #include <string>
 
 #include "lo-math.h"
-#include "systime.h"
 
 class octave_base_tm;
 
 class
 OCTAVE_API
 octave_time
 {
 public:
diff --git a/liboctave/systime.h b/liboctave/systime.h
deleted file mode 100644
--- a/liboctave/systime.h
+++ /dev/null
@@ -1,37 +0,0 @@
-/*
-
-Copyright (C) 1996, 1997, 1999, 2005, 2007 John W. Eaton
-
-This file is part of Octave.
-
-Octave is free software; you can redistribute it and/or modify it
-under the terms of the GNU General Public License as published by the
-Free Software Foundation; either version 3 of the License, or (at your
-option) any later version.
-
-Octave is distributed in the hope that it will be useful, but WITHOUT
-ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
-FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
-for more details.
-
-You should have received a copy of the GNU General Public License
-along with Octave; see the file COPYING.  If not, see
-<http://www.gnu.org/licenses/>.
-
-*/
-
-#if !defined (octave_systime_h)
-#define octave_systime_h 1
-
-#ifdef TIME_WITH_SYS_TIME
-#include <sys/time.h>
-#include <time.h>
-#else
-#ifdef HAVE_SYS_TIME_H
-#include <sys/time.h>
-#else
-#include <time.h>
-#endif
-#endif
-
-#endif
diff --git a/src/ChangeLog b/src/ChangeLog
--- a/src/ChangeLog
+++ b/src/ChangeLog
@@ -1,8 +1,17 @@
+2010-02-02  John W. Eaton  <jwe@octave.org>
+
+	* cutils.c: Include <sys/times.h>, not "systime.h".
+	* data.cc (Fcputime): Assume we have times.
+
+	* DLD-FUNCTIONS/getrusage.cc (Fgetrusage): Assume we have times.
+	Include <times.h> and <sys/times.h> unconditionally.
+	Don't include <windows.h>
+
 2010-02-01  Jaroslav Hajek  <highegg@gmail.com>
 
 	* strfns.cc (Fstrncmpi): Fix typo in docstring.
 
 2010-01-30  Jaroslav Hajek  <highegg@gmail.com>
 
 	* ov-list.h, ov-list.cc, OPERATORS/op-list.cc: Remove.
 
diff --git a/src/DLD-FUNCTIONS/getrusage.cc b/src/DLD-FUNCTIONS/getrusage.cc
--- a/src/DLD-FUNCTIONS/getrusage.cc
+++ b/src/DLD-FUNCTIONS/getrusage.cc
@@ -20,58 +20,45 @@ along with Octave; see the file COPYING.
 <http://www.gnu.org/licenses/>.
 
 */
 
 #ifdef HAVE_CONFIG_H
 #include <config.h>
 #endif
 
-#include "systime.h"
-
+#include <sys/time.h>
+#include <sys/times.h>
 #include <sys/types.h>
 
 #ifdef HAVE_SYS_RESOURCE_H
 #include <sys/resource.h>
 #endif
 
-#if defined (__WIN32__)
-#include <windows.h>
-#ifdef min
-#undef min
-#undef max
-#endif
-#endif
-
-#if defined (HAVE_TIMES) && defined (HAVE_SYS_TIMES_H)
-
 #if defined (HAVE_SYS_PARAM_H)
 #include <sys/param.h>
 #endif
-#include <sys/times.h>
+
+#include "defun-dld.h"
+#include "oct-map.h"
+#include "sysdep.h"
+#include "ov.h"
+#include "oct-obj.h"
+#include "utils.h"
 
 #if !defined (HZ)
 #if defined (CLK_TCK)
 #define HZ CLK_TCK
 #elif defined (USG)
 #define HZ 100
 #else
 #define HZ 60
 #endif
 #endif
 
-#endif
-
-#include "defun-dld.h"
-#include "oct-map.h"
-#include "sysdep.h"
-#include "ov.h"
-#include "oct-obj.h"
-#include "utils.h"
-
 #ifndef RUSAGE_SELF
 #define RUSAGE_SELF 0
 #endif
 
 // System resource functions.
 
 DEFUN_DLD (getrusage, , ,
   "-*- texinfo -*-\n\
@@ -167,17 +154,16 @@ elements @code{sec} (seconds) @code{usec
   m.assign ("msgsnd", static_cast<double> (ru.ru_msgsnd));
   m.assign ("msgrcv", static_cast<double> (ru.ru_msgrcv));
   m.assign ("nsignals", static_cast<double> (ru.ru_nsignals));
   m.assign ("nvcsw", static_cast<double> (ru.ru_nvcsw));
   m.assign ("nivcsw", static_cast<double> (ru.ru_nivcsw));
 #endif
 
 #else
-#if defined (HAVE_TIMES) && defined (HAVE_SYS_TIMES_H)
 
   struct tms t;
 
   times (&t);
 
   unsigned long ticks;
   unsigned long seconds;
   unsigned long fraction;
@@ -193,42 +179,16 @@ elements @code{sec} (seconds) @code{usec
   ticks = t.tms_stime + t.tms_cstime;
   fraction = ticks % HZ;
   seconds = ticks / HZ;
 
   tv_tmp.assign ("sec", static_cast<double> (seconds));
   tv_tmp.assign ("usec", static_cast<double> (fraction * 1e6 / HZ));
   m.assign ("stime", octave_value (tv_tmp));
 
-#elif defined (__WIN32__)
-  HANDLE hProcess = GetCurrentProcess ();
-  FILETIME ftCreation, ftExit, ftUser, ftKernel;
-  GetProcessTimes (hProcess, &ftCreation, &ftExit, &ftKernel, &ftUser);
-
-  int64_t itmp = *(reinterpret_cast<int64_t *> (&ftUser));
-  tv_tmp.assign ("sec", static_cast<double> (itmp / 10000000U));
-  tv_tmp.assign ("usec", static_cast<double> (itmp % 10000000U) / 10.);
-  m.assign ("utime", octave_value (tv_tmp));
-
-  itmp = *(reinterpret_cast<int64_t *> (&ftKernel));
-  tv_tmp.assign ("sec", static_cast<double> (itmp / 10000000U));
-  tv_tmp.assign ("usec", static_cast<double> (itmp % 10000000U) / 10.);
-  m.assign ("stime", octave_value (tv_tmp));
-#else
-
-  tv_tmp.assign ("sec", 0);
-  tv_tmp.assign ("usec", 0);
-  m.assign ("utime", octave_value (tv_tmp));
-
-  tv_tmp.assign ("sec", 0);
-  tv_tmp.assign ("usec", 0);
-  m.assign ("stime", octave_value (tv_tmp));
-
-#endif
-
   double tmp = lo_ieee_nan_value ();
 
   m.assign ("maxrss", tmp);
   m.assign ("ixrss", tmp);
   m.assign ("idrss", tmp);
   m.assign ("isrss", tmp);
   m.assign ("minflt", tmp);
   m.assign ("majflt", tmp);
diff --git a/src/cutils.c b/src/cutils.c
--- a/src/cutils.c
+++ b/src/cutils.c
@@ -30,21 +30,20 @@ along with Octave; see the file COPYING.
 #include <string.h>
 
 #if defined (__WIN32__) && ! defined (_POSIX_VERSION)
 
 #include <windows.h>
 
 #else
 
+#include <sys/time.h>
 #include <sys/types.h>
 #include <unistd.h>
 
-#include "systime.h"
-
 #ifdef HAVE_POLL_H
 #include <poll.h>
 #elif HAVE_SYS_POLL_H
 #include <sys/poll.h>
 #endif
 
 #endif
 
diff --git a/src/data.cc b/src/data.cc
--- a/src/data.cc
+++ b/src/data.cc
@@ -22,30 +22,29 @@ along with Octave; see the file COPYING.
 <http://www.gnu.org/licenses/>.
 
 */
 
 #ifdef HAVE_CONFIG_H
 #include <config.h>
 #endif
 
-#include "systime.h"
-
 #include <sys/types.h>
 
 #ifdef HAVE_SYS_RESOURCE_H
 #include <sys/resource.h>
 #endif
 
 #include <cfloat>
 
 #include <string>
 
 #include "lo-ieee.h"
 #include "lo-math.h"
+#include "oct-time.h"
 #include "str-vec.h"
 #include "quit.h"
 #include "mx-base.h"
 
 #include "Cell.h"
 #include "defun.h"
 #include "error.h"
 #include "gripes.h"
@@ -5662,17 +5661,17 @@ CPU time used is nonzero.\n\
   getrusage (RUSAGE_SELF, &ru);
 
   usr = static_cast<double> (ru.ru_utime.tv_sec) +
     static_cast<double> (ru.ru_utime.tv_usec) * 1e-6;
 
   sys = static_cast<double> (ru.ru_stime.tv_sec) +
     static_cast<double> (ru.ru_stime.tv_usec) * 1e-6;
 
-#elif defined (HAVE_TIMES) && defined (HAVE_SYS_TIMES_H)
+#else
 
   struct tms t;
 
   times (&t);
 
   unsigned long ticks;
   unsigned long seconds;
   unsigned long fraction;
